src/core/coap/coap_secure.cpp

```diff
@@ -244,13 +274,10 @@ void CoapSecure::HandleTransmit(void)
 exit:
     if (error != OT_ERROR_NONE)
     {
-        otLogNoteMeshCoP("CoapSecure Transmit: %s", otThreadErrorToString(error));
         message->Free();
     }
-    else
-    {
-        otLogDebgMeshCoP("CoapSecure Transmit: %s", otThreadErrorToString(error));
-    }
+
+    otLogNoteMeshCoP("CoapSecure Transmit: %s", otThreadErrorToString(error));
```

Comment:
Can we keep the original logging? `NOTE` level is intended for normal but significant events that require special handling (e.g., device getting detached) and are expected to be less frequent than INFO and DEBG level.